apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( {
+			isTemplateModalOpen: ! this.state.isTemplateModalOpen,
+			isWarningOpen: false,
+		} );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = () => {
+		if ( ! this.props.templateUsedSlug || this.props.templateUsedSlug === 'blank' ) {
+			return this.props.templates[ 1 ];
```

Comment:
Is the blank template always in this slot or should we check for it more specifically?

Reply:
Actually, from what I can tell the blank template is always in index 0.  Since 'Blank' isn't really a template and it has no real image for previewing, I have set it to index 1 to default to preview some template that actually represents a template (as far as i can tell index 1 seems to always be the Home template)

Reply:
I guess that raises the question why `this.props.templateUsedSlug === 'blank'` shouldn't reference the blank template. 🤔